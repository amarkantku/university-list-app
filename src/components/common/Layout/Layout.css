.row {  
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
}

.main {
    display: block;
    margin: auto;
    padding: 20px;
    background-color: #fff;
    min-width: 75%;
    min-height: 80vh;
}

.heading {
    margin: auto;
    height: auto;
    padding: 10px 0;
    text-decoration: underline;
}
