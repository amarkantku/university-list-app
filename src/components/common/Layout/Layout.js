import React from 'react';
import classNames from 'classnames/bind';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import styles from './Layout.css';
let clx = classNames.bind(styles);

export default function Layout(props) {
    return (
        <div data-testid='layout' className={clx('layout', 'row')}>
            <Header data-testid='header'/>
            <main data-testid='main' className={clx('main')}>
                <h2 className={clx('heading')}>{props.title}</h2>
                {props.children}
            </main>
            <Footer />
        </div>
    )
}

