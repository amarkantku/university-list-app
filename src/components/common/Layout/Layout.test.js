import React from 'react';
import Layout from './Layout';
import { render } from '@testing-library/react';


const defaultProps = {
    title: 'Test Layout Title'
}
describe("<Layout />", () => {
    test ('should component render', () => {
        const { asFragment } = render(<Layout {...defaultProps} />);
        expect(asFragment()).toMatchSnapshot();
    });
      
    test('should have getByTestId="layout"', () => {
        const { getByTestId } = render(<Layout {...defaultProps} />);
        expect(getByTestId('layout')).toBeTruthy();
    });

    test('should have getByTestId="main"', () => {
        const { getByTestId } = render(<Layout {...defaultProps} />);
        expect(getByTestId('main')).toBeTruthy();
    });

    test('should have contain h2 header text ="Test Layout Title"', () => {
        const { queryByText } = render(<Layout {...defaultProps} />);
        expect(queryByText(defaultProps.title)).toBeTruthy();
    });

    test('should have children', () => {
        const { queryByText } = render(<Layout {...defaultProps}>Children</Layout>);
        expect(queryByText('Children')).toBeTruthy();
    });
})