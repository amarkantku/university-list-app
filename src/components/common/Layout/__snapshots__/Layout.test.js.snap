// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should component render 1`] = `
<DocumentFragment>
  <div
    class="layout row"
    data-testid="layout"
  >
    <header
      data-testid="header"
    >
      <div
        class="header"
      >
        <h1>
          University List App
        </h1>
      </div>
    </header>
    <main
      class="main"
      data-testid="main"
    >
      <h2
        class="heading"
      >
        Test Layout Title
      </h2>
    </main>
    <footer
      data-testid="footer"
    >
      <div
        class="footer"
      >
        Footer
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
